streaming/base/dataset.py

```diff
@@ -97,7 +97,8 @@ def __init__(self,
                  timeout: float = 60,
                  hash: Optional[str] = None,
                  batch_size: Optional[int] = None) -> None:
-        keep_zip = (remote == local) if keep_zip is None else keep_zip
+        if keep_zip is None:
```

Comment:
I see that `keep_zip: Optional[bool] = None,`, should we have `False` as a default? E.g.: `keep_zip: Optional[bool] = False,` ?

Reply:
Explainer:

Let's assume zipping provides net benefit (for example, text data), so zipping is used for a given dataset. Shards will therefore exist in two forms: the zipped form which they are stored as, and the unzipped form which they are used as. Remote directories store in only the zipped form, and local directories need to store only the unzipped form. So normally, you'd want keep_zip = False, to conserve storage as locals don't need the zipped form which they are downloaded as. But there's a special case: what if there is no remote/the local is the remote? In that case, you want to not blow away the zipped shards because they are how the dataset is persistently stored. The default value of keep_zip = None does that ("False, except that special case").

Reply:
Make sense. Thanks for the explanation!